cpp/dolfinx/fem/interpolate.h

```diff
@@ -269,44 +300,44 @@ void interpolate_nonmatching_maps(Function<T>& u1, const Function<T>& u0,
     {
       const int pos = 3 * x_dofs[i];
       for (std::size_t j = 0; j < gdim; ++j)
-        coordinate_dofs(i, j) = x_g[pos + j];
+        coord_dofs(i, j) = x_g[pos + j];
     }
 
     // Compute Jacobians and reference points for current cell
-    J.fill(0);
-    for (std::size_t p = 0; p < X.shape(0); ++p)
+    std::fill(J_b.begin(), J_b.end(), 0);
+    for (std::size_t p = 0; p < Xshape[0]; ++p)
     {
-      auto _J = xt::view(J, p, xt::all(), xt::all());
-      cmap.compute_jacobian(dphi, coordinate_dofs, _J);
-      cmap.compute_jacobian_inverse(_J, xt::view(K, p, xt::all(), xt::all()));
-      detJ[p] = cmap.compute_jacobian_determinant(_J);
+      auto _J = stdex::submdspan(J, p, stdex::full_extent, stdex::full_extent);
+      cmap.compute_jacobian(dphi, coord_dofs, _J);
```

Comment:
I think there is a similar bug here for non-affine elements, `dphi` refers to the derivatives at the first interpolation point, see line 240. This subspan should be moved into the loop. 

I guess we have too few tests on non-affine meshes.

Reply:
True, but since it's not a 'new' bug I'd rather fix it in a subsequent PR. It's much easier now to see what's happening to we should review all the relevant code. See also #2275.